src/test/java/com/sawyereffect/cucumber/documentation/ConfluenceMarkupFormatTest.java

```diff
@@ -0,0 +1,122 @@
+        when(annotationType1.name()).thenReturn("When");
+        when(annotationType2.name()).thenReturn("When");
+        when(annotationType3.name()).thenReturn("Then");
+
+        when(annotationDesc1.annotationType()).thenReturn(annotationType1);
+        when(annotationDesc2.annotationType()).thenReturn(annotationType2);
+        when(annotationDesc3.annotationType()).thenReturn(annotationType3);
+
+        when(annotationDesc1.elementValues()).thenReturn(elementValues1);
+        when(annotationDesc2.elementValues()).thenReturn(elementValues2);
+        when(annotationDesc3.elementValues()).thenReturn(elementValues3);
+
+        when(paramTag.parameterName()).thenReturn("searchTerm");
+        when(paramTag.parameterComment()).thenReturn("- term to be searched.");
+
+        when(paramTag3.parameterName()).thenReturn("results");
+        when(paramTag3.parameterComment()).thenReturn("- text to be searched for validation.");
+
+        when(methodDoc1.commentText()).thenReturn("Executes a search given the term provided on the search page.");
+        when(methodDoc1.name()).thenReturn("she_enters");
+        when(methodDoc1.annotations()).thenReturn(annotationsDesc1);
+        when(methodDoc1.paramTags()).thenReturn(paramTags);
+
+        when(methodDoc2.commentText()).thenReturn("User hits the first result available.");
+        when(methodDoc2.name()).thenReturn("click_on_the_first_link_from_text_results");
+        when(methodDoc2.annotations()).thenReturn(annotationsDesc2);
+        when(methodDoc2.paramTags()).thenReturn(paramTags2);
+
+        when(methodDoc3.commentText()).thenReturn("Verify from results given an specific text is displayed.");
+        when(methodDoc3.name()).thenReturn("results_should_be_displayed");
+        when(methodDoc3.annotations()).thenReturn(annotationsDesc3);
+        when(methodDoc3.paramTags()).thenReturn(paramTags3);
+
+        when(classDoc.methods()).thenReturn(methodDocs);
+
+        confluenceMarkupFormat.parse(classDoc, fileWriter);
+
+        verify(fileWriter).write("\\\\\n" +
+                "h2. com.sawyereffect.steps.SearchPageSteps\n*Google search page*\n" +
+                " This page is used to search for terms in the outside world.\n||ACTIONS||\n" +
+                "||Method||Step||Description||\n" +
+                "|she_enters|^She enters ([^\"]*)$|Executes a search given the term provided on the search page.\n" +
+                "@param searchTerm - term to be searched.|\n" +
+                "|click_on_the_first_link_from_text_results|^Click on the first link from text results$|User hits the first result available.|\n" +
+                "\\\\\n" +
+                "||VERIFICATIONS||\n" +
+                "||Method||Step||Description||\n" +
+                "|results_should_be_displayed|^Verify ([^\"]*) results should be displayed$|Verify from results given an specific text is displayed.\n" +
+                "@param results - text to be searched for validation.|\n");
+
+    }
+
```

Comment:
Well, I was describing what it is doing but I am not sure if that's what we want it to do. @npgarcia What do you think it would be best to do in case there are no comments and also when there are no annotations of an specific kind in a class (i.e. No Verifications in a page)? Shall we display the header empty or no to display it at all?